flink-ml-lib/src/test/java/org/apache/flink/ml/stats/ChiSqTestTest.java

```diff
@@ -0,0 +1,205 @@
+        inputTableWithIntegerLabel =
+                tEnv.fromDataStream(env.fromCollection(samplesWithIntegerLabel))
+                        .as("label", "f1", "f2");
+        inputTableWithStringLabel =
+                tEnv.fromDataStream(env.fromCollection(samplesWithStringLabel))
+                        .as("label", "f1", "f2");
+    }
+
+    private static void verifyPredictionResult(Table output, List<Row> expected) throws Exception {
+        StreamTableEnvironment tEnv =
+                (StreamTableEnvironment) ((TableImpl) output).getTableEnvironment();
+        DataStream<Row> outputDataStream = tEnv.toDataStream(output);
+
+        List<Row> result = IteratorUtils.toList(outputDataStream.executeAndCollect());
+
+        compareResultCollections(
+                expected,
+                result,
+                (row1, row2) -> {
+                    if (!row1.equals(row2)) {
+                        return 1;
+                    } else {
+                        return 0;
+                    }
+                });
+    }
+
+    @Test
+    public void testParam() {
+        ChiSqTest chiSqTest = new ChiSqTest();
+
+        chiSqTest.setInputCols("f1", "f2").setLabelCol("label");
+
+        assertArrayEquals(new String[] {"f1", "f2"}, chiSqTest.getInputCols());
+        assertEquals("label", chiSqTest.getLabelCol());
+    }
+
+    @Test
+    public void testOutputSchema() {
+        ChiSqTest chiSqTest = new ChiSqTest().setInputCols("f1", "f2").setLabelCol("label");
+
+        Table output = chiSqTest.transform(inputTableWithDoubleLabel)[0];
+        assertEquals(
+                Arrays.asList("column", "pValue", "statistic", "degreesOfFreedom"),
+                output.getResolvedSchema().getColumnNames());
+    }
+
+    @Test
+    public void testTransform() throws Exception {
+        ChiSqTest chiSqTest = new ChiSqTest().setInputCols("f1", "f2").setLabelCol("label");
+
+        Table output1 = chiSqTest.transform(inputTableWithDoubleLabel)[0];
+        verifyPredictionResult(output1, expectedChiSqTestResultWithDoubleLabel);
+
+        Table output2 = chiSqTest.transform(inputTableWithIntegerLabel)[0];
+        verifyPredictionResult(output2, expectedChiSqTestResultWithIntegerLabel);
+
+        Table output3 = chiSqTest.transform(inputTableWithStringLabel)[0];
+        verifyPredictionResult(output3, expectedChiSqTestResultWithStringLabel);
+    }
+
+    @Test
+    public void testSaveLoadAndTransform() throws Exception {
+        ChiSqTest chiSqTest = new ChiSqTest().setInputCols("f1", "f2").setLabelCol("label");
+
+        ChiSqTest loadedBucketizer =
+                TestUtils.saveAndReload(tEnv, chiSqTest, tempFolder.newFolder().getAbsolutePath());
+        Table output1 = loadedBucketizer.transform(inputTableWithDoubleLabel)[0];
+        verifyPredictionResult(output1, expectedChiSqTestResultWithDoubleLabel);
+
+        Table output2 = loadedBucketizer.transform(inputTableWithIntegerLabel)[0];
```

Comment:
nit: I think one input for `testSaveLoadAndTransform` is enough since different inputs are already tested in `testTransform`.